apps/antalmanac/src/components/RightPane/CoursePane/CoursePaneButtonRow.tsx

```diff
@@ -89,28 +94,31 @@ export function ColumnToggleButton() {
     return (
         <>
             <Tooltip title="Show/Hide Columns">
-                <IconButton onClick={handleOpen} sx={buttonSx}>
+                <IconButton onClick={handleClick} sx={buttonSx}>
                     <Visibility />
                 </IconButton>
             </Tooltip>
-            <FormControl>
-                <Select
-                    multiple
-                    value={selectedColumnNames}
-                    open={open}
-                    onChange={handleChange}
-                    onClose={handleClose}
-                    renderValue={renderEmptySelectValue}
-                    sx={{ visibility: 'hidden', position: 'absolute' }}
-                >
-                    {COLUMN_LABEL_ENTRIES.map(([column, label], index) => (
-                        <MenuItem key={column} value={column}>
-                            <Checkbox checked={selectedColumns[index]} color="default" />
-                            <ListItemText primary={label} />
-                        </MenuItem>
-                    ))}
-                </Select>
-            </FormControl>
+
+            <Popover open={open} anchorEl={anchorEl} onClose={handleClose} sx={{ visibility: 'hidden' }}>
+                <FormControl>
+                    <Select
+                        multiple
+                        value={selectedColumnNames}
+                        open={open}
+                        onChange={handleChange}
+                        onClose={handleClose}
+                        renderValue={renderEmptySelectValue}
+                        MenuProps={{ anchorEl }}
```

Comment:
This is related to some Select bug, loosely documented and described in [a chain of duplicate/related Github Issues](https://github.com/mui/material-ui/issues/19245)...

TL;DR anchorEl makes it work like a charm